WWW/smoorthi-dev/scatter_gather/minidata_mutation_calling_v3.wdl

```diff
@@ -238,58 +246,68 @@ task MarkDuplicates {
     }
 
 
-
 task splitBambyChr {
-  input{
+  input {
     File bamtosplit
+    File baitosplit
     Array[String] chromosomes
-  }
-  
+    }
+    String base_file_name = basename(bamtosplit, "_final.duplicates_marked.bam")
   command <<<
-  set -eo pipefail
-  for x in "${chromosomes[@]}"
-  do
-  # Create BAM file
-  samtools view -b -@ 3 ~{bamtosplit} ${x} > ${x}.bam
-  # Create index file
-  samtools index ${x}.bam
-  done
-  >>>
-  output{
-    Array[File] bams = glob("*.bam")
-    Array[File] indexFiles = glob("*.bam.bai")
-  }
+    set -eo pipefail
+    baseFileName="~{base_file_name}"
   
+    for x in ~{sep=' ' chromosomes}; do
+    outputFile="${baseFileName}_${x}.bam"
```

Comment:
I prefer to have everything within a bash for loop indented, because that makes it very clear what is and is not within that loop. But that might be my Python bias.